[qunat][graphmode][fx] Standalone module takes float as input and output

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48671 [qunat][graphmode][fx] Standalone module takes float as input and output**

Summary:
Standalone module might be called separately so it's better to use float
as interface.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25256184](https://our.internmc.facebook.com/intern/diff/D25256184)

jerryzh168 committed:
[qunat][graphmode][fx] Standalone module takes float as input and output

Summary:
Standalone module might be called separately so it's better to use float
as interface.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[qunat][graphmode][fx] Standalone module takes float as input and output"


Summary:
Standalone module might be called separately so it's better to use float
as interface.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25256184](https://our.internmc.facebook.com/intern/diff/D25256184)

[ghstack-poisoned]

vkuzo reviewed (approved):
Accepting to unblock.  I'd love to learn more about the motivations.  

Do I understand correctly that for an arbitrary node, we observe the output and the weights (if applicable), with the input observation coming from the preceding node.  If yes, would this PR move standalone module farther from how other nodes work?

Would it make sense to in the future treat standalone module as a regular node, allowing for the user to specify whether inputs and outputs need to be in fp32 or quantized?

jerryzh168 commented:
> Do I understand correctly that for an arbitrary node, we observe the output and the weights (if applicable), with the input observation coming from the preceding node. If yes, would this PR move standalone module farther from how other nodes work?

yes, this is correct, previously it works this way, basically standalone module is quantized in a similar way as nn.quantized.Conv2d, it takes quantized tensor as input and produces quantized output.

However, standalone module can also be used in other cases, e.g. called directly in the code, and in that case it will be better to make it self contained and use float -> float interface.


> Would it make sense to in the future treat standalone module as a regular node, allowing for the user to specify whether inputs and outputs need to be in fp32 or quantized?

I think we can allow that if there is a use case.

jerryzh168 committed:
Update on "[qunat][graphmode][fx] Standalone module takes float as input and output"


Summary:
Standalone module might be called separately so it's better to use float
as interface.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25256184](https://our.internmc.facebook.com/intern/diff/D25256184)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[qunat][graphmode][fx] Standalone module takes float as input and output"


Summary:
Standalone module might be called separately so it's better to use float
as interface.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25256184](https://our.internmc.facebook.com/intern/diff/D25256184)

[ghstack-poisoned]
